src/lib/pickles/test/ec-scaling-foreign/test.ml

```diff
@@ -0,0 +1,345 @@
+open Core_kernel
+open Pickles.Impls.Step
+
+module type S = sig
+  type foreign
+
+  val foreign_typ : (foreign, unit) Typ.t
+
+  val zero : foreign
+
+  val assert_equal : foreign -> foreign -> unit
+
+  val range_check : foreign -> unit
+
+  val add_chain : range_check:(foreign -> unit) -> foreign -> foreign -> foreign
```

Comment:
The `range_check` argument is always omitted in the modules implementing this signature. 

Either change the signature for `add_chain` or explain why this argument needs to be here (I haven't found other uses for this signature outside of this module).

Reply:
This is just a prototype. I was leaving it open in case we decided that we also need the range checks on the results of additions.

I'm not planning to merge this, it has no value, except potentially as a reference for the proper implementation later.